### PR TITLE
Infrastructure for creating user

### PR DESCRIPTION
**Pull request by profileEnumerable**

### CODE REVIEW

File: Src/Campus.Infrastructure.Business/Services/ProfileService.cs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+        public async Task<ProfileViewModelDto> GetAppUserProfileByIdAsync(int id)
+        {
+            _unitOfWork.Begin();
+
+            try
+            {
+                var appUser = await _appUserRepository.GetAppUserByIdAsync(id);
+
+                _unitOfWork.Commit();
```
Comment (antonAce) [inline Src/Campus.Infrastructure.Business/Services/ProfileService.cs:54]: There aren't any WRITE operations to the DB, you can remove transaction mechanisms

File: Src/Campus.Infrastructure.Business/Services/ProfileService.cs [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            catch (Exception)
+            {
+                _unitOfWork.Rollback();
+                throw;
+            }
+        }
+
+        public async Task<int> DeleteAppUserProfileByIdAsync(int id)
+        {
+            return await _appUserRepository.DeleteAppUserByIdAsync(id);
```
Comment (antonAce) [inline Src/Campus.Infrastructure.Business/Services/ProfileService.cs:73]: You should commit changes after DELETE operations

File: Src/Campus.Infrastructure.Business/Services/ProfileService.cs [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        }
+
+        public async Task<int> EditAppUserProfileByIdAsync(int id, ProfileEditingModelDto editingDto)
+        {
+            return await _appUserRepository.UpdateAppUserAsync(new AppUser()
```
Comment (antonAce) [inline Src/Campus.Infrastructure.Business/Services/ProfileService.cs:78]: You should commit changes after UPDATE operations

File: Src/Campus.Master.API/Startup.cs [hunk 4]
```diff
@@ -31,25 +32,35 @@ public Startup(IConfiguration configuration)
         // This method gets called by the runtime. Use this method to add services to the container.
         public void ConfigureServices(IServiceCollection services)
         {
-            var xmlDocFile = $"{Assembly.GetExecutingAssembly().GetName().Name}.xml";
-            var xmlDocPath = Path.Combine(AppContext.BaseDirectory, xmlDocFile);
-            
+            string xmlDocFile = $"{Assembly.GetExecutingAssembly().GetName().Name}.xml";
```
Comment (antonAce) [inline Src/Campus.Master.API/Startup.cs:35]: Why did you change VAR to STRING?

File: Src/Campus.Infrastructure.Data/packages.config [hunk 5]
```diff
@@ -0,0 +1,4 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<packages>
+  <package id="Dapper" version="2.0.35" targetFramework="net48" />
```
Comment (antonAce) [inline Src/Campus.Infrastructure.Data/packages.config:3]: Change target to .net core 3.1 for all projects

File: Src/Campus.Infrastructure.Data/Repositories/UnitOfWork.cs [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+        public void Begin()
+        {
+            _transaction = _connection.BeginTransaction();
+        }
+
+        public void Commit()
+        {
+            _transaction.Commit();
+            Dispose();
```
Comment (antonAce) [inline Src/Campus.Infrastructure.Data/Repositories/UnitOfWork.cs:29]: DI container performs object dispose after every scoped execution, you don't need to do it manually

File: Src/Campus.Infrastructure.Data/Repositories/UnitOfWork.cs [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+        }
+
+        public void Dispose()
+        {
+            if (_transaction != null)
+            {
+                _transaction.Dispose();
+                _connection.Close();
+            }
+            _transaction = null;
```
Comment (antonAce) [inline Src/Campus.Infrastructure.Data/Repositories/UnitOfWork.cs:45]: I need explanations)

File: Src/Campus.Domain.Core/Models/AppUser.cs [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+        
+        public string Name { get; set; }
+
+        public string Surname { get; set; }
+
+        public string Email { get; set; }
+
+        public string Login { get; set; }
+
+        public string PasswordHash { get; set; }
```
Comment (antonAce) [inline Src/Campus.Domain.Core/Models/AppUser.cs:15]: Using password hash is not enough, you need to add password salt to strengthen auth security

File: Src/Campus.Master.API/Startup.cs [hunk 9]
```diff
@@ -31,25 +32,35 @@ public Startup(IConfiguration configuration)
...
+                c.AddSecurityRequirement(new OpenApiSecurityRequirement
+                {
+                    {new OpenApiSecurityScheme {Name = "Bearer"}, Enumerable.Empty<string>().ToList()},
                 });
-                
+
                 c.IncludeXmlComments(xmlDocPath);
             });
+            
+            services.AddScoped<IDbConnection>(c => new SqlConnection(Configuration["ConnectionStrings:Default"]));
```
Comment (antonAce) [inline Src/Campus.Master.API/Startup.cs:59]: You shouldn't register service explicitly like this, because you're dragging dependencies of lower layers to the higher. Write service providing extension methods for the Data and Business layers. Basically, we'll get rid of unused dependencies in the API layer like SQL server dependencies. Check out extension method example here: https://github.com/antonAce/onion-di/blob/master/src/OnionDI/OnionDI.Data/Dependencies/DataServicesExtensions.cs

File: Src/Campus.Master.API/Startup.cs [hunk 10]
```diff
@@ -31,25 +32,35 @@ public Startup(IConfiguration configuration)
...
+            services.AddScoped<IUnitOfWork, UnitOfWork>();
+            services.AddScoped<IAppUserRepository, AppUserRepository>();
+            
+            services.AddScoped<IProfileService, ProfileService>();
```
Comment (antonAce) [inline Src/Campus.Master.API/Startup.cs:63]: Also requires extension methods

File: Src/Campus.Master.API/appsettings.json [hunk 11]
```diff
@@ -9,5 +9,8 @@
   "AllowedHosts": "*",
   "Endpoints": {
     "QueryItemsLimit": 20
+  },
+  "ConnectionStrings": {
+    "Default": "Data Source=(LocalDB)\\MSSQLLocalDB;Database=CampusDb;Integrated Security=True;MultipleActiveResultSets=True"
```
Comment (antonAce) [inline Src/Campus.Master.API/appsettings.json:14]: Change also connection strings for the dev appsettings file. And remove explicit connection strings declaration from prod mode file, because it's sensitive information, we'll keep it in secrets manager

File: Src/Campus.Infrastructure.Business/Services/ProfileService.cs (hunk 1 above)
Comment (profileEnumerable) [reply Src/Campus.Infrastructure.Business/Services/ProfileService.cs:54]: Removed

File: Src/Campus.Infrastructure.Business/Services/ProfileService.cs (hunk 2 above)
Comment (profileEnumerable) [reply Src/Campus.Infrastructure.Business/Services/ProfileService.cs:73]: Added

File: Src/Campus.Infrastructure.Business/Services/ProfileService.cs (hunk 3 above)
Comment (profileEnumerable) [reply Src/Campus.Infrastructure.Business/Services/ProfileService.cs:78]: Added

File: Src/Campus.Master.API/Startup.cs (hunk 4 above)
Comment (profileEnumerable) [reply Src/Campus.Master.API/Startup.cs:35]: My code conventions settings have been applied

Comment (profileEnumerable) [reply Src/Campus.Master.API/Startup.cs:35]: This settings change var to string in non-obvious cases

File: Src/Campus.Infrastructure.Data/packages.config (hunk 5 above)
Comment (profileEnumerable) [reply Src/Campus.Infrastructure.Data/packages.config]: Changed

File: Src/Campus.Master.API/Startup.cs (hunk 4 above)
Comment (antonAce) [reply Src/Campus.Master.API/Startup.cs:35]: Wierd conventions, but I'll let you keep 'em for this time)

File: Src/Campus.Master.API/Startup.cs (hunk 9 above)
Comment (profileEnumerable) [reply Src/Campus.Master.API/Startup.cs]: Done

File: Src/Campus.Master.API/Startup.cs (hunk 10 above)
Comment (profileEnumerable) [reply Src/Campus.Master.API/Startup.cs]: Done

File: Src/Campus.Master.API/appsettings.json (hunk 11 above)
Comment (profileEnumerable) [reply Src/Campus.Master.API/appsettings.json]: Moved
